komga/src/main/kotlin/org/gotson/komga/domain/service/BookLifecycle.kt

```diff
@@ -272,4 +281,23 @@ class BookLifecycle(
       eventPublisher.publishEvent(DomainEvent.ReadProgressDeleted(progress))
     }
   }
+
+  fun deleteBookFiles(book: Book) {
+    if (book.path.notExists() || !book.path.isWritable())
+      throw FileNotFoundException("File is not accessible : ${book.path}").withCode("ERR_1018")
+
+    val thumbnails = thumbnailBookRepository.findAllByBookId(book.id)
+      .filter { it.type == ThumbnailBook.Type.SIDECAR }
+      .mapNotNull { it.url?.toURI()?.toPath() }
+
+    thumbnails.find { it.notExists() || !it.isWritable() }?.let {
+      throw FileNotFoundException("File is not accessible : ${book.path}").withCode("ERR_1018")
+    }
```

Comment:
that means the book deletion fails if you can't find one of the sidecar ? That's drastic, sidecars are updated only during scans, so we don't really care if they are there or not. If they are, we delete them, if not, the Thumbnail entry will get cleaned up eventually.